public/app/features/alerting/unified/components/rule-editor/util.ts

```diff
@@ -261,6 +260,9 @@ export function getThresholdsForQueries(queries: AlertQuery[]) {
     // now also sort the threshold values, if we don't then they will look weird in the time series panel
     // TODO this doesn't work for negative values for now, those need to be sorted inverse
     thresholds[refId].config.steps.sort((a, b) => a.value - b.value);
+
+    // also make sure we remove any "undefined" values from our steps in case the threshold config is incomplete
+    thresholds[refId].config.steps = thresholds[refId].config.steps.filter((step) => step.value !== undefined);
```

Comment:
is this change needed for this PR changes? or you added as an improvement?

Reply:
Not strictly need, it fixes a small bug I found while testing my changes where an incomplete threshold definition (like `is between 0 and <undefined>` would break the time series visualization and show a blank canvas.